BuildResidentialScheduleFile/tests/build_residential_schedule_file_test.rb

```diff
@@ -180,6 +180,51 @@ def test_stochastic
     assert(!sf.schedules.keys.include?(SchedulesFile::ColumnVacancy))
   end
 
+  def test_stochastic_subset_of_columns
+    hpxml = _create_hpxml('base.xml')
+    XMLHelper.write_file(hpxml.to_oga, @tmp_hpxml_path)
+
+    columns = [SchedulesFile::ColumnCookingRange,
+               SchedulesFile::ColumnDishwasher,
+               SchedulesFile::ColumnHotWaterDishwasher,
+               SchedulesFile::ColumnClothesWasher,
+               SchedulesFile::ColumnHotWaterClothesWasher,
+               SchedulesFile::ColumnClothesDryer,
+               SchedulesFile::ColumnHotWaterFixtures]
+
+    @args_hash['schedules_type'] = 'stochastic'
+    @args_hash['output_csv_path'] = File.absolute_path(File.join(@tmp_output_path, 'occupancy-stochastic.csv'))
+    @args_hash['schedules_column_names'] = columns.join(', ')
+    model, hpxml, result = _test_measure()
+
+    info_msgs = result.info.map { |x| x.logMessage }
+    assert(info_msgs.any? { |info_msg| info_msg.include?('ColumnNames') })
+
+    sf = SchedulesFile.new(model: model, schedules_paths: hpxml.header.schedules_filepaths)
+    sf.validate_schedules(year: 2007)
+
+    columns.each do |column|
+      assert(sf.schedules.keys.include?(column))
+    end
+    (SchedulesFile.ColumnNames - columns).each do |column|
+      assert(!sf.schedules.keys.include?(column))
+    end
+  end
+
+  def test_stochastic_subset_of_columns_invalid_name
+    hpxml = _create_hpxml('base.xml')
+    XMLHelper.write_file(hpxml.to_oga, @tmp_hpxml_path)
+
+    @args_hash['schedules_type'] = 'stochastic'
+    @args_hash['output_csv_path'] = File.absolute_path(File.join(@tmp_output_path, 'occupancy-stochastic.csv'))
+    @args_hash['schedules_column_names'] = 'foobar, foobar2'
```

Comment:
What if you have at least one valid column name along with an invalid column name? Does the whole thing error out, or does the invalid column name just get ignored?

Reply:
The whole thing errors out. I will update the test to demonstrate that.